src/r_util.cpp

```diff
@@ -441,6 +441,42 @@ namespace rhost {
             return result;
         }
 
+        extern "C" SEXP create_blob(SEXP obj) {
+            int type = TYPEOF(obj);
+            size_t length = Rf_length(obj);
+
+            if (type != RAWSXP || length == 0) {
+                Rf_error("Object must be a RAW vector of length > 0.");
+            }
+
+            Rbyte* data = RAW(obj);
+            rhost::util::blob blob;
+            blob.push_back(rhost::util::blob_slice(data, data + length));
+            int id = rhost::host::create_blob(blob);
+            return Rf_ScalarInteger((int)id);
+        }
+
+        extern "C" SEXP get_blob(SEXP id) {
+            int blob_id = Rf_asInteger(id);
+            const std::vector<byte> data = rhost::host::get_blob_as_single_slice(blob_id);
+
+            if (data.size() == 0) {
+                return R_NilValue;
+            }
+
+            SEXP rawVector = Rf_allocVector(RAWSXP, data.size());
```

Comment:
This probably needs to be `Rf_protect`-ed - I don't think `RAW` can potentially trigger a GC, but I don't think it guarantees it, either.